modules/core/04-channel/keeper/upgrade.go

```diff
@@ -355,11 +355,19 @@ func (k Keeper) ChanUpgradeOpen(
 		if !found {
 			return errorsmod.Wrapf(types.ErrUpgradeNotFound, "failed to retrieve channel upgrade: port ID (%s) channel ID (%s)", portID, channelID)
 		}
+		// If counterparty has reached OPEN, we must use the upgraded connection to verify the counterparty channel
+		upgradedConnection, err := k.GetConnection(ctx, upgrade.Fields.ConnectionHops[0])
+		if err != nil {
+			return errorsmod.Wrap(err, "failed to retrieve connection using the upgrade connection hops")
+		}
+		if upgradedConnection.GetState() != int32(connectiontypes.OPEN) {
+			return errorsmod.Wrapf(connectiontypes.ErrInvalidConnectionState, "connection state is not OPEN (got %s)", connectiontypes.State(upgradedConnection.GetState()).String())
+		}
```

Comment:
some nits:

Would be nice to add some whitespacing between if's etc, let the code breathe! 🧘‍♀️ 

I'm not sure how I feel about `upgradedConnection` var naming, as the connection itself isn't/hasn't been upgraded. I'd be happy with just `connection` or maybe `upgradeConnection`

Reply:
yeah, connection conflicted with earlier connection defined, upgradeConnection it is.

Reply:
lol @ breathing code